### PR TITLE
feat(FEC-11600): Related grid analytics

### PR DESCRIPTION
**Pull request by SivanA-Kaltura**
### Description of the Changes

Fire events on toggle button click and on manual entry selection
Expose event types for kava

Related PR: https://github.com/kaltura/playkit-js-kava/pull/133

Resolves FEC-11600

### CheckLists

- [ ] changes have been done against master branch, and PR does not conflict
- [ ] new unit / functional tests have been added (whenever applicable)
- [ ] test are passing in local environment
- [ ] Travis tests are passing (or test results are not worse than on master branch :))
- [ ] Docs have been updated

### CODE REVIEW

File: src/related.tsx [hunk 1]
```diff
@@ -120,10 +120,10 @@ class Related extends KalturaPlayer.core.BasePlugin {
     const preplayBackPlayOverlayProps = {
       relatedManager,
       onLoaded: (callback: (nextEntries: []) => void) => {
-        relatedManager.listen(RelatedEvent.HIDDEN_STATE_CHANGED, ({payload}: {payload: []}) => callback(payload));
+        relatedManager.listen(RelatedInternalEvent.HIDDEN_STATE_CHANGED, ({payload}: {payload: []}) => callback(payload));
       },
       onUnloaded: (cb: (nextEntries: []) => void) => {
-        relatedManager.unlisten(RelatedEvent.HIDDEN_STATE_CHANGED, cb);
+        relatedManager.unlisten(RelatedInternalEvent.HIDDEN_STATE_CHANGED, cb);
```
Comment (JonathanTGold) [inline src/related.tsx:126]: Please use the event manager decorator and call removeAll on onUnmount hook

File: src/related.tsx [hunk 2]
```diff
@@ -120,10 +120,10 @@ class Related extends KalturaPlayer.core.BasePlugin {
     const preplayBackPlayOverlayProps = {
       relatedManager,
       onLoaded: (callback: (nextEntries: []) => void) => {
-        relatedManager.listen(RelatedEvent.HIDDEN_STATE_CHANGED, ({payload}: {payload: []}) => callback(payload));
+        relatedManager.listen(RelatedInternalEvent.HIDDEN_STATE_CHANGED, ({payload}: {payload: []}) => callback(payload));
```
Comment (JonathanTGold) [inline src/related.tsx:123]: Please use the event manager decorator and call removeAll on onUnmount hook

Don't use the same event manager of the plugin

**Review by JonathanTGold — CHANGES_REQUESTED** [review]

File: src/related.tsx (hunk 1 above)
Comment (SivanA-Kaltura) [reply src/related.tsx:126]: as we discussed, fixing it on the other PR https://github.com/kaltura/playkit-js-related/pull/33

File: src/related.tsx (hunk 2 above)
Comment (SivanA-Kaltura) [reply src/related.tsx:123]: as we discussed, fixing it on the other PR https://github.com/kaltura/playkit-js-related/pull/33

File: src/types/events.js [hunk 3]
```diff
@@ -0,0 +1,37 @@
+/**
+ * Related plugin events which are dispatched externally through the player.
+ */
+const RelatedEvent = {
```
Comment (JonathanTGold) [inline src/types/events.js:4]: this is a model not a type so i think it should not be in types/ directory , i suggest to create a new file called related-events

**Review by JonathanTGold — CHANGES_REQUESTED** [review]

File: src/types/events.js (hunk 3 above)
Comment (SivanA-Kaltura) [reply src/types/events.js:4]: i've moved it to another folder in this PR:
https://github.com/kaltura/playkit-js-related/pull/37/files#diff-2a70666c28d275ea3d772c26f80af76782534e0feae34c3b483e409dfc87ee9f

**Review by giladna — APPROVED** [review]

**Review by JonathanTGold — APPROVED** [review]

**Pull request merged**